### PR TITLE
Add API: IsDefaultProtocolClient

### LINKED ISSUE #5164: Add `isDefaultProtocolClient` API
The new `setAsDefaultProtocolClient`  and `removeAsDefaultProtocolClient` APIs are great, but without a way to query the current state, it's not possible to implement a UI element like this one. I think it'd be great to add `isDefaultProtocolClient`!

image

Also, I find the naming of these APIs pretty confusing. On Mac OS X and iOS `mailto` is referred to as a "URL Scheme", not a protocol (https://developer.apple.com/library/ios/featuredarticles/iPhoneURLScheme_Reference), and it seems like that might be the case on Windows too. (https://msdn.microsoft.com/en-us/library/aa767914(v=vs.85).aspx)

It'd be nice to clarify that this is registering the app system-wide, while the other `protocol` module is only for handling requests initiated from within the app. 

Maybe we could change these to `registerForURLScheme`, `isRegisteredForURLScheme`?

### PR DESCRIPTION
**Pull request by ritazh**
@zcbenz @felixrieseberg This should fix #5164

### CODE REVIEW

File: atom/browser/browser_win.cc [hunk 1]
```diff
@@ -225,6 +225,50 @@ bool Browser::SetAsDefaultProtocolClient(const std::string& protocol) {
...
+  std::wstring keyVal;
+  if (FAILED(commandKey.ReadValue(L"", &keyVal)))
+    // Default value not set, we can confirm that it is not set
+    return false;
+
+  std::wstring exePath(path.value());
+  std::wstring exe = L"\"" + exePath + L"\" \"%1\"";
+  if (keyVal == exe) {
+    // Default value is the same as current file path
+      return true;
```
Comment (felixrieseberg) [inline atom/browser/browser_win.cc]: Bonus spaces 😸

**Comment by felixrieseberg** [comment]
:party: Sweet! I found two little extra spaces, but otherwise, this looks 👍

File: atom/browser/browser_mac.mm [hunk 2]
```diff
@@ -60,6 +60,25 @@
...
+    return false;
+
+  NSString* protocol_ns = [NSString stringWithUTF8String:protocol.c_str()];
+  
+  CFStringRef bundle =
+      LSCopyDefaultHandlerForURLScheme(base::mac::NSToCFCast(protocol_ns));
+  if (!bundle){
+    return false;
+  }
+  NSString* bundleId = base::mac::CFToNSCast(bundle);
```
Comment (zcbenz) [inline atom/browser/browser_mac.mm]: The `bundle` is a `CFStringRef` so it has to be released, you can reference how Chromium did it:
https://code.google.com/p/chromium/codesearch#chromium/src/chrome/browser/shell_integration_mac.mm&sq=package:chromium&type=cs&l=34

Comment (ritazh) [inline atom/browser/browser_mac.mm]: Thanks for the pointer @zcbenz. 👏  Code updated.

**Comment by zcbenz** [comment]
Thanks!

**Pull request merged**